src/bin/cql-stress-cassandra-stress/java_generate/values/hex_blob.rs

```diff
@@ -0,0 +1,113 @@
+use super::ValueGenerator;
+use crate::java_generate::distribution::Distribution;
+
+/// Based on c-s HexBytes generator.
+/// Used by c-s to generate blob partition keys.
+/// TBH, not quite sure why c-s won't use Bytes generator for this.
+pub struct HexBlob();
+
+impl ValueGenerator for HexBlob {
+    type Value = Vec<u8>;
+
+    fn generate(
+        &mut self,
+        identity_distribution: &mut dyn Distribution,
+        size_distribution: &mut dyn Distribution,
+    ) -> Self::Value {
+        let seed = identity_distribution.next_i64();
+        size_distribution.set_seed(seed);
+        let size = size_distribution.next_i64() as usize;
+
+        let mut result = vec![0; size];
+        let mut i = 0;
+
+        // No idea what's going on here.
+        // Check https://github.com/scylladb/scylla-tools-java/blob/master/tools/stress/src/org/apache/cassandra/stress/generate/values/HexBytes.java#L44
+        while i < size {
+            let mut value = identity_distribution.next_i64();
+            let mut j = 0;
+            while j < 16 && i + j < size {
+                let v = (value & 15) as i32;
+                result[i + j] = ((if v < 10 { '0' as i32 } else { 'A' as i32 }) + v) as u8;
+                // Unsigned right shift.
+                value = ((value as u64) >> 4) as i64;
+                j += 1;
+            }
+            i += 16;
+        }
+
+        result
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::java_generate::{
+        distribution::{fixed::FixedDistribution, sequence::SeqDistribution, Distribution},
+        values::{Generator, GeneratorConfig},
+    };
+
+    use super::HexBlob;
+
+    fn to_vec_i8(v: Vec<u8>) -> Vec<i8> {
+        v.into_iter().map(|x| x as i8).collect::<Vec<i8>>()
+    }
+
+    #[test]
+    /// This test is based on cassandra-stress HexBytes generator.
+    /// The command cassandra-stress was being run with:
+    /// write n=5 no-warmup -node 172.17.0.2 -pop seq=1..5 -col size=FIXED(5) -rate threads=1
+    fn hex_blob_generator_test() {
+        // Generator of partition key column (by default named "key").
+        let config = GeneratorConfig::new(
+            // "Random" string used to generate the generator's salt.
+            // In c-s, it's always "randomstr" + `column_name`.
+            // See https://github.com/scylladb/scylla-tools-java/blob/master/tools/stress/src/org/apache/cassandra/stress/settings/SettingsCommandPreDefined.java#L86
+            String::from("randomstrkey"),
+            None,
+            // By default the keysize=10
+            Some(Box::new(FixedDistribution::new(10))),
+        );
+        let hex_blob = HexBlob();
+        let mut gen = Generator::new(hex_blob, config);
+
+        // -pop seq=1..5
+        // Samples from this distrubtion are the seeds to the partition key generator.
+        let mut seq = SeqDistribution::new(1, 5).unwrap();
```

Comment:
This length might be a bit too small to check a scenario where `while i < size` in `generate()` loops twice.